transformer.py

```diff
@@ -52,6 +57,98 @@
 object_store = None
 
 
+def initialize_logging(request=None):
+    """
+    Get a logger and initialize it so that it outputs the correct format
+
+    :param request: Request id to insert into log messages
+    :return: logger with correct formatting that outputs to console
+    """
+
+    log = logging.getLogger()
+    if 'INSTANCE_NAME' in os.environ:
+        instance = os.environ['INSTANCE_NAME']
+    else:
+        instance = 'Unknown'
+    formatter = logging.Formatter('%(levelname)s ' +
+                                  "{} xaod_cpp_transformer {} ".format(instance, request) +
+                                  '%(message)s')
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    handler.setLevel(logging.INFO)
+    log.addHandler(handler)
+    log.setLevel(logging.INFO)
+    return log
+
+
+def parse_output_logs(logfile):
+    """
+    Parse output from runner.sh and output appropriate log messages
+    :param logfile: path to logfile
+    :return:  None
+    """
+    total_events = 0
+    events_processed = 0
+    total_events_re = re.compile(r'Processing events \d+-(\d+)')
+    events_processed_re = re.compile(r'Processed (\d+) events')
+    with open(logfile, 'r') as f:
+        buf = f.read()
+        match = total_events_re.search(buf)
+        if match:
+            total_events = int(match.group(1))
+        matches = events_processed_re.finditer(buf)
+        for m in matches:
+            events_processed = int(m.group(1))
+        logger.info("{} events processed out of {} total events".format(events_processed, total_events))
+
+
+# class TimeTuple(NamedTuple):
+class TimeTuple(namedtuple("TimeTupleInit", ["user", "system", "idle"])):
+    """
+    Named tuple to store process time information.
+    Immutable so values can't be accidentally altered after creation
+    """
+    # user: float
+    # system: float
+    # idle: float
+
+    @property
+    def total_time(self):
+        """
+        Return total time spent by process
+
+        :return: sum of user, system, idle times
+        """
+        return self.user + self.system + self.idle
+
+
+def get_process_info():
+    """
+    Get process information (just cpu, sys, idle times right now) and return it
+
+    :return: TimeTuple with timing information
+    """
+    time_stats = psutil.cpu_times()
+    return TimeTuple(user=time_stats.user, system=time_stats.system, idle=time_stats.idle)
+
+
+def log_stats(startup_time, total_time, running_time=0.0):
```

Comment:
Would be nice to have type annotations for `startup_time` and `total_time`.

Reply:
Likewise, this code runs under python 2.7 so type annotations are a no go without moving to the really clumsy type annotations in docstring style.